torch/nn/parallel/replicate.py

```diff
@@ -180,4 +183,4 @@ def replicate(
                     replica = module_copies[j][i]
                     setattr(replica, key, buffer_copies[j][buffer_idx])
 
-    return [module_copies[j][0] for j in range(num_replicas)]
+    return [cast(T, module_copies[j][0]) for j in range(num_replicas)]
```

Comment:
Why not change the type of module_copies to `LIst[List[T]]`?

Reply:
If you need to cast it too, you could just cast the entire list instead of casting each element, right?

Reply:
Only the first element of `module_copies` (the first element of `network.modules()`) has type `T`:

```python
reveal_type(network)  # Type of "network" is "T@replicate"
modules = list(network.modules())
module_copies: List[List[Module]] = [[] for _ in devices]
...

for i, module in enumerate(modules):
    ...
    for j in range(num_replicas):
        replica = module._replicate_for_data_parallel()
        ...
        module_copies[j].append(replica)

```

Reply:
Ah, I guess it makes since in this context then. The proper way to do it would be an assert with an isinstance check, but that is annoying to do in a listcomp so feel free to leave it as is.